src/Raven.Server/Documents/PeriodicBackup/RavenFtpClient.cs

```diff
@@ -62,63 +70,68 @@ public void UploadFile(string folderName, string fileName, Stream stream)
             Progress?.UploadProgress.SetTotal(stream.Length);
             Progress?.UploadProgress.ChangeState(UploadState.PendingUpload);
 
-            var url = CreateNestedFoldersIfNeeded(folderName);
-            url += $"/{fileName}";
+            var url = CreateNestedFoldersIfNeeded(folderName, out string path);
+            path += $"/{fileName}";
 
-            var request = CreateFtpWebRequest(url, WebRequestMethods.Ftp.UploadFile, keepAlive: true);
-            var readBuffer = new byte[DefaultBufferSize];
-
-            int count;
-            var requestStream = request.GetRequestStream();
-            while ((count = stream.Read(readBuffer, 0, readBuffer.Length)) != 0)
+            using (var client = CreateFtpClient(url, keepAlive: true))
             {
-                requestStream.Write(readBuffer, 0, count);
-
-                Progress?.UploadProgress.ChangeState(UploadState.Uploading);
-                Progress?.UploadProgress.UpdateUploaded(count);
-                Progress?.OnUploadProgress();
-            }
-
-            requestStream.Flush();
-            requestStream.Close();
+                try
+                {
+                    var readBuffer = new byte[DefaultBufferSize];
 
-            Progress?.UploadProgress.ChangeState(UploadState.PendingResponse);
-            using (request.GetResponse())
-            {
-                Progress?.UploadProgress.ChangeState(UploadState.Done);
+                    int count;
+                    while ((count = stream.Read(readBuffer, 0, readBuffer.Length)) != 0)
+                    {
+                        client.UploadBytes(readBuffer, path, FtpRemoteExists.Resume, progress: (p) =>
```

Comment:
how does the progress behave? you have a while loop here and start to upload another chunk, wouldn't it set the value to 0 again? can't it be done via 1 call?

Reply:
Done